src/wallet/wallet.cpp

```diff
@@ -5186,14 +5186,21 @@ void CWallet::AvailableCoins(vector<COutput>& vCoins,
             if (nDepth < nMinDepth)
                 continue;
 
+            // if coincontrol allows watch-only addresses, add ISMINE_WATCH_SOLVABLE to the filter
+            const auto watchConditionedFilter =
+                (coinControl != nullptr && coinControl->fAllowWatchOnly) ?
+                    (filter | ISMINE_WATCH_SOLVABLE) :
+                    filter;
+
             for (unsigned int i = 0; i < pcoin.vout.size(); i++) {
                 const auto& output = pcoin.vout[i];
                 isminetype mine = IsMine(output);
 
-                bool isSpendable = ((mine & ISMINE_SPENDABLE) != ISMINE_NO) ||
-                                    (coinControl && coinControl->fAllowWatchOnly && (mine & ISMINE_WATCH_SOLVABLE) != ISMINE_NO);
+                CTxDestination addr;
+                ExtractDestination(output.scriptPubKey, addr);
 
-                if (fOnlySpendable && !isSpendable)
+                bool consideredSpendable = (mine & watchConditionedFilter) != ISMINE_NO;
+                if (!consideredSpendable)
```

Comment:
```suggestion
                if (fOnlySpendable && !consideredSpendable)
```
This fixes the issue I identified earlier, because now `consideredSpendable` can be false and still make it into the output.